contracts/dapis/DapiServer.sol

```diff
@@ -806,6 +811,145 @@ contract DapiServer is
+    /// Airnodes without requiring a request or subscription. The Beacons for
+    /// which the signature is omitted will be read from the storage.
+    /// @param airnodes Airnode addresses
+    /// @param templateIds Template IDs
+    /// @param timestamps Timestamps used in the signatures
+    /// @param data Response data (a `uint256` encoded in contract ABI per
+    /// Beacon)
+    /// @param metadata Metadata that the OEV proxy will use (e.g., auction bid
+    /// parameters)
+    /// @param signatures Template ID, a timestamp, response data and metadata
+    /// signed by the respective Airnode address per Beacon
+    /// @return beaconSetId Beacon set ID
+    function updateOevProxyBeaconSetWithSignedData(
+        address[] memory airnodes,
+        bytes32[] memory templateIds,
+        uint256[] memory timestamps,
+        bytes[] memory data,
+        bytes memory metadata,
+        bytes[] memory signatures
+    ) external override returns (bytes32 beaconSetId) {
+        uint256 beaconCount = airnodes.length;
+        require(
+            beaconCount == templateIds.length &&
+                beaconCount == timestamps.length &&
+                beaconCount == data.length &&
+                beaconCount == signatures.length,
+            "Parameter length mismatch"
+        );
+        require(beaconCount > 1, "Specified less than two Beacons");
+        bytes32[] memory beaconIds = new bytes32[](beaconCount);
+        uint256[] memory values = new uint256[](beaconCount);
+        uint256 accumulatedTimestamp = 0;
+        for (uint256 ind = 0; ind < beaconCount; ind++) {
+            if (signatures[ind].length != 0) {
+                uint256 timestamp = timestamps[ind];
+                require(timestampIsValid(timestamp), "Timestamp not valid");
+                require(
+                    (
+                        keccak256(
+                            abi.encodePacked(
+                                templateIds[ind],
+                                timestamp,
+                                data[ind],
+                                metadata
+                            )
+                        ).toEthSignedMessageHash()
+                    ).recover(signatures[ind]) == airnodes[ind],
+                    "Signature mismatch"
+                );
+                values[ind] = decodeFulfillmentData(data[ind]);
+                // Timestamp validity is already checked, which means it will
+                // be small enough to be typecast into `uint32`
+                accumulatedTimestamp += timestamp;
+                beaconIds[ind] = deriveBeaconId(
+                    airnodes[ind],
+                    templateIds[ind]
+                );
+            } else {
+                bytes32 beaconId = deriveBeaconId(
+                    airnodes[ind],
+                    templateIds[ind]
+                );
+                DataFeed storage dataFeed = dataFeeds[beaconId];
```

Comment:
We read from `dataFeeds[beaconId]` and not `oevProxyAddressToIdToDataFeed[msg.sender][beaconId]`. That's because if a proxy is pointing to a Beacon set, it's constituent Beacons will not receive any OEV updates and `oevProxyAddressToIdToDataFeed[msg.sender][beaconId]` is expected to be empty, while `dataFeeds[beaconId]` may receive individual updates to be used as a fallback.